services/movie_session.py

```diff
@@ -0,0 +1,53 @@
+from db.models import MovieSession
+import datetime
+from django.db.models import QuerySet
+from typing import Optional
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int,
+) -> QuerySet:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id,
+    )
+
+
+def get_movies_sessions(session_date: Optional[str] = None) -> QuerySet:
+    if session_date is not None:
+        date_object = datetime.datetime.strptime(session_date, "%Y-%m-%d")
+        return MovieSession.objects.filter(show_time__date=date_object)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_by_id: int) -> QuerySet:
+    return MovieSession.objects.get(id=movie_session_by_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: Optional[int] = None,
+        cinema_hall_id: Optional[int] = None,
+) -> QuerySet:
+    queryset = MovieSession.objects.all()
+    if show_time is not None:
+        queryset = MovieSession.objects.filter(

```

Comment:
no need to again use MovieSession.objects.filter
use instead queryset = queryset.filter(

Reply:
The same in other places